deepchem/models/torch_models/layers.py

```diff
@@ -5674,43 +5669,49 @@ def forward(self, one_electron: torch.Tensor,
+            det = []
             for i in range(self.spin[0]):
                 one_d_index = (k * (self.total_electron)) + i
                 for j in range(self.spin[0]):
-                    psi_up[:, k, i, j] = (torch.sum(
+                    det.append(((torch.sum(
                         (self.envelope_w[one_d_index] * one_electron[:, j, :]) +
                         self.envelope_g[one_d_index],
                         dim=1)) * torch.sum(torch.exp(-torch.abs(
-                            torch.norm(self.sigma[one_d_index] *
-                                       one_electron_vector_permuted[:, j, :, :],
+                            torch.norm(one_electron_vector_permuted[:, j, :, :]
+                                       @ self.sigma[one_d_index],
                                        dim=2))) * self.pi[one_d_index].T,
-                                            dim=1)
+                                            dim=1)))
+            psi_up.append(
+                torch.reshape(torch.stack(det, dim=1),
+                              (self.batch_size, self.spin[0], self.spin[0])))
 
+            det = []
             for i in range(self.spin[0], self.spin[0] + self.spin[1]):
                 one_d_index = (k * (self.total_electron)) + i
                 for j in range(self.spin[0], self.spin[0] + self.spin[1]):
-                    psi_down[:, k, i - self.spin[0], j - self.spin[0]] = (
-                        torch.sum((self.envelope_w[one_d_index] *
-                                   one_electron[:, j, :]) +
-                                  self.envelope_g[one_d_index],
-                                  dim=1)
-                    ) * torch.sum(torch.exp(-torch.abs(
-                        torch.norm(self.sigma[one_d_index] *
-                                   one_electron_vector_permuted[:, j, :, :],
-                                   dim=2))) * self.pi[one_d_index].T,
-                                  dim=1)
-
-            d_down = torch.det(psi_down[:, k, :, :].clone())
-            d_up = torch.det(psi_up[:, k, :, :].clone())
-            det = d_up * d_down
-            psi = psi + det
-        return psi, psi_up, psi_down
+                    det.append(((torch.sum(
+                        (self.envelope_w[one_d_index] * one_electron[:, j, :]) +
+                        self.envelope_g[one_d_index],
+                        dim=1)) * torch.sum(torch.exp(-torch.abs(
+                            torch.norm(one_electron_vector_permuted[:, j, :, :]
+                                       @ self.sigma[one_d_index],
+                                       dim=2))) * self.pi[one_d_index].T,
+                                            dim=1)))
+            psi_down.append(
+                torch.reshape(torch.stack(det, dim=1),
+                              (self.batch_size, self.spin[1], self.spin[1])))
+
+            d_down = torch.det(psi_down[-1])
+            d_up = torch.det(psi_up[-1])
```

Comment:
earlier version was not working for single electron ions, refactored the code accordingly to make changes to work for all systems